### ISSUE TITLE
Video location is not returned

### ISSUE THREAD

**Issue by BarryTPL** [aimmac23/selenium-video-node#38]
Hello,

My current setup is WebDriver 2.48.2 and selenium video node 2.3. The hub is a Linux machine with no UI and the node is a Windows server 2012 R2 machine.

I started the hub with the command below, specifying to store video file locally (i.e. /tmp).
`nohup java -cp selenium-video-node-2.3.jar:selenium-server-standalone-2.48.2.jar org.openqa.grid.selenium.GridLauncher -servlets com.aimmac23.hub.servlet.HubVideoDownloadServlet,com.aimmac23.hub.servlet.HubVideoInfoServlet -Dvideo.storage=com.aimmac23.hub.videostorage.LocalFileVideoStore -Dvideo.path=/tmp -role hub hubConfig hubconfig.json &`

After the test case is finished, which can be confirmed by the log messages below:
`10:11:51.712 INFO - Selenium session closed for c8c887fb-95b4-485e-b003-d33ce3642bd1 on node http://<node ip>:5597 - stopping recording.
10:11:52.271 INFO - Successfully retrieved video for session: c8c887fb-95b4-485e-b003-d33ce3642bd1 and temporarily stashed it at: /tmp/screencast2698924965919350547.webm`

I use the HubVideoInfoServlet to retrieve information about the video, but the path of the video is not included in the response.
`http://<hub ip>:4444/grid/admin/HubVideoInfoServlet/?sessionId=c8c887fb-95b4-485e-b003-d33ce3642bd1`

As you can see, additional is empty.
`{
    "additional": {},
    "fileSize": 46085,
    "storageType": "TEMP_FILE"
}`

The video is created and playable, the location returned by the log message is correct. Am I missing anything else here?

I tried without using -Dvideo.storage and -Dvideo.path but the outcome is the same.

I have a few questions regarding the video storage location and I hope you can shed some light on them.

- The video is first recorded in the node then transferred to the hub right?
- -Dvideo.path specifies the storage location on the hub, not the node correct?
- I tried to specify a different location other than /tmp on the hub, but videos are still saved at /tmp, am I doing anything wrong?
- Is there any way to capture the log message that contains the file location such as this one?
`10:11:52.271 INFO - Successfully retrieved video for session: c8c887fb-95b4-485e-b003-d33ce3642bd1 and temporarily stashed it at: /tmp/screencast2698924965919350547.webm`

Thanks,
Barry

**Comment by BarryTPL** [comment]
After some investigation, I found out that I need to put double quotes around the class name in order for the hub to recognize it, just like issue 21.

This is a bit different from the documentation and I guess it might be because I am calling it in a pre-build jenkins script.

Barry

**Comment by aimmac23** [comment]
Hi BarryTPL,

I'm glad you got this working. I'm not aware of any issue where double quotes are required for the classname - I'll experiment with it this weekend so I can try to reproduce it. What language is your script written in?

Thanks,

Alasdair

**Comment by BarryTPL** [comment]
Hi Alasdair,

My scripts are all java based.

I have a script executed in the jenkins job to start the hub under Build > Execute managed script.

Within the script, the command I used to start the hub is:
nohup java -Dvideo.storage="com.aimmac23.hub.videostorage.LocalFileVideoStore" -Dvideo.path=${SCREENSHOT_FOLDER} -cp selenium-video-node-2.3.jar:selenium-server-standalone-2.48.2.jar org.openqa.grid.selenium.GridLauncher -servlets com.aimmac23.hub.servlet.HubVideoInfoServlet,com.aimmac23.hub.servlet.HubVideoDownloadServlet -role hub hub hubConfig hubconfig.json &

Before I added double quotes around the video storage class, the hub always starts with the default value "com.aimmac23.hub.videostorage.LocalTempFileStore".

Thanks,
Barry
